scripts/run-external.sh

```diff
@@ -1,20 +1,273 @@
 #!/usr/bin/env bash
-# exit immediately when a command fails
-set -e
-# only exit with zero if all commands of the pipeline exit successfully
-set -o pipefail
-# error on unset variables
-set -u
-
-if [[ -z "${1:-}" ]]; then
-	echo "ERROR: missing cluster name"
-	echo "usage: ❯ $0 <clustername>"
-	exit 1
-fi
-
-apiserver=$(kubectl config view -o jsonpath="{.clusters[?(@.name == \"$1\")].cluster.server}")
-cafile=$(kubectl config view -o jsonpath="{.clusters[?(@.name == \"$1\")].cluster.certificate-authority}")
-certfile=$(kubectl config view -o jsonpath="{.users[?(@.name == \"$1\")].user.client-certificate}")
-keyfile=$(kubectl config view -o jsonpath="{.users[?(@.name == \"$1\")].user.client-key}")
-
-./operator --apiserver="${apiserver}" --ca-file="${cafile}" --cert-file="${certfile}" --key-file="${keyfile}"
+# NOTE:
+#  -e exit immediately when a command fails
+#  -u error on unset variables
+#  -o pipefail  exit with zero only if all commands of the pipeline exit successfully
+set -eu -o pipefail
+
+trap cleanup EXIT INT
+
+# globals
+declare CLUSTER=""
+declare SHOW_USAGE=false
+declare SKIP_OPERATOR_RUN_CHECK=false
+declare USE_DEFAULT_CONTEXT=false
+declare API_SERVER=""
+
+# tmp operator files that needs to be cleaned up
+declare -r CA_FILE="tmp/CA_FILE"
+declare -r KEY_FILE="tmp/KEY_FILE"
+declare -r CERT_FILE="tmp/CERT_FILE"
+
+cleanup() {
+	rm -f "$CA_FILE" "$KEY_FILE" "$CERT_FILE"
+}
+
+run() {
+	echo " ❯ $* "
+	"$@"
+}
+
+header() {
+	local title="🔆🔆🔆  $*  🔆🔆🔆 "
+
+	local len=40
+	if [[ ${#title} -gt $len ]]; then
+		len=${#title}
+	fi
+
+	echo -e "\n\n  \033[1m${title}\033[0m"
+	echo -n "━━━━━"
+	printf '━%.0s' $(seq "$len")
+	echo "━━━━━"
+}
+
+info() {
+	echo " 🔔 $*"
+}
+
+ok() {
+	echo " ✅ $*"
+}
+
+warn() {
+	echo " ⚠️  $*"
+}
+
+err() {
+	echo " 🛑 ERROR: $*"
+}
+
+update_crds() {
+	header "Update Operator CRDs"
+
+	run kubectl apply --server-side --force-conflicts \
+		-f example/prometheus-operator-crd-full
+
+	info "Wait for CRDs to be registered"
+	run kubectl wait --for=condition=Established crds --all --timeout=120s
+}
+
+init_cluster_context() {
+	header "Set Cluster"
+	$USE_DEFAULT_CONTEXT && {
+		CLUSTER=$(kubectl config current-context)
+	}
+
+	info "Using cluster - '$CLUSTER'"
+	return 0
+}
+
+extract_certs() {
+	local cluster=".clusters[?(@.name == \"$CLUSTER\")].cluster"
+	local user=".users[?(@.name == \"$CLUSTER\")].user"
+
+	local ca key cert
+
+	API_SERVER=$(kubectl config view -o jsonpath="{$cluster.server}")
+	ca=$(kubectl config view --raw -o jsonpath="{$cluster.certificate-authority-data} | base64 -d")
```

Comment:
sorry my initial suggestion had a typo
```suggestion
	ca=$(kubectl config view --raw -o jsonpath="{$cluster.certificate-authority-data}" | base64 -d)
```